src/Bridges/slackbridge.jl

```diff
@@ -0,0 +1,162 @@
+end
+function VectorSlackBridge{T, F, S}(model, f::F, s::S) where {T, F<:MOI.AbstractVectorFunction, S<:MOI.AbstractVectorSet}
+    d = MOI.dimension(s)
+    slacks = MOI.add_variables(model, d)
+    new_f = MOIU.operate(-, T, f, MOI.VectorAffineFunction{T}(MOI.VectorOfVariables(slacks)))
+    slacks_in_set = MOI.add_constraint(model, MOI.VectorOfVariables(slacks), s)
+    equality = MOI.add_constraint(model, new_f, MOI.Zeros(d))
+    return VectorSlackBridge{T, F, S}(slacks, slacks_in_set, equality)
+end
+
+MOI.supports_constraint(::Type{VectorSlackBridge{T, F, S}},
+                        ::Type{<:MOI.AbstractVectorFunction},
+                        ::Type{<:MOI.AbstractVectorSet}) where {T, F, S} = true
+MOI.supports_constraint(::Type{VectorSlackBridge{T, F, S}},
+                        ::Type{<:MOI.VectorOfVariables},
+                        ::Type{<:MOI.Zeros}) where {T, F, S} = false
+MOI.supports_constraint(::Type{VectorSlackBridge{T, F, S}},
+                        ::Type{<:MOI.AbstractVectorFunction},
+                        ::Type{<:MOI.Zeros}) where {T, F, S} = false
+MOI.supports_constraint(::Type{VectorSlackBridge{T, F, S}},
+                        ::Type{<:MOI.VectorOfVariables},
+                        ::Type{<:MOI.AbstractVectorSet}) where {T, F, S} = false
+function added_constraint_types(::Type{VectorSlackBridge{T, F, S}}) where {T, F<:MOI.AbstractVectorFunction, S}
+    return [(F, MOI.Zeros), (MOI.VectorOfVariables, S)]
+end
+function concrete_bridge_type(::Type{<:VectorSlackBridge{T}},
+                              F::Type{<:MOI.AbstractVectorFunction},
+                              S::Type{<:MOI.AbstractVectorSet}) where T
+    return VectorSlackBridge{T, F, S}
+end
+
+# Attributes, Bridge acting as an model
+MOI.get(b::VectorSlackBridge{T, F, S}, ::MOI.NumberOfConstraints{F, MOI.Zeros}) where {T, F<:MOI.AbstractVectorFunction, S} = 1
+MOI.get(b::VectorSlackBridge{T, F, S}, ::MOI.NumberOfConstraints{MOI.VectorOfVariables, S}) where {T, F<:MOI.AbstractVectorFunction, S} = 1
+MOI.get(b::VectorSlackBridge{T, F, S}, ::MOI.ListOfConstraintIndices{F, MOI.Zeros}) where {T, F<:MOI.AbstractVectorFunction, S} = [b.equality]
+MOI.get(b::VectorSlackBridge{T, F, S}, ::MOI.ListOfConstraintIndices{MOI.VectorOfVariables, S}) where {T, F<:MOI.AbstractVectorFunction, S} = [b.slacks_in_set]
+
+# Indices
+function MOI.delete(model::MOI.ModelLike, c::VectorSlackBridge)
+    MOI.delete(model, c.equality)
+    MOI.delete(model, c.slacks_in_set)
+    MOI.delete(model, c.slacks)
+    return
+end
+
+# Attributes, Bridge acting as a constraint
+function MOI.get(model::MOI.ModelLike, attr::MOI.ConstraintPrimal, c::VectorSlackBridge)
+    # due to equality, slacks should have the same value as original affine function
+    return MOI.get(model, attr, c.slacks_in_set)
+end
+function MOI.get(model::MOI.ModelLike, a::MOI.ConstraintDual, c::VectorSlackBridge)
+    error("VectorSlackBridge is not returning duals for now")
```

Comment:
The dual constraint on `slack` (since it is free) is `-dual_slack_in_set  + dual_equality = 0` so the two duals are equal and we can return either one of them. I would return the dual of `slack_in_set`